tests/test_cursor_async.py

```diff
@@ -270,10 +270,9 @@ async def test_iter_stop(aconn):
 
 async def test_row_factory(aconn):
     def my_row_factory(cursor):
-        assert cursor.description is not None
-        titles = [c.name for c in cursor.description]
-
         def mkrow(values):
+            assert cursor.description is not None
+            titles = [c.name for c in cursor.description]
```

Comment:
This change invalidates the test: the row_factory is not called again on pgresult switch in nextset() and this change hides it. Fixed in 7d341d0

Reply:
I made this change¹ because I was thinking that the row factory should in fact not be called again on `nextset()` but only the row maker should. In other words, I assumed that the "lifetime" of the row maker (the function returned by the row factory) was that of the cursor, whereas it should be that of a result set, if I understand correctly?

¹: and should have mentioned it, sorry

Reply:
I understand, no worries :) No, a cursor can manage more than one result in its lifetime: either using execute() more than once (which was already handled correctly), or using more than one query and scrolling them with `nextset()`: in these cases results are independent. The other case, the one I fixed in master and merge here, is that _sometimes_ it's guaranteed that the new result object has the same columns of the previous one (in `stream()` there is one result per row, in server-side cursors there is one result per `FETCH`), so I have created a code path where the per-columns-set initialisation is not repeated. I did it to to skip instantiating the `Loader` objects for the data conversion, but it's exactly the same code path we can use to e.g. avoiding creating a new NamedTuple class if the previous one was already good.